packages/integration/src/gitlab/GitLabIntegration.ts

```diff
@@ -62,6 +81,92 @@ export class GitLabIntegration implements ScmIntegration {
   resolveEditUrl(url: string): string {
     return replaceGitLabUrlType(url, 'edit');
   }
+
+  fetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response> {
+    return this.fetchImpl(input, init);
+  }
+
+  private createFetchStrategy(): FetchFunction {
+    let fetchFn: FetchFunction = async (url, options) => {
+      return fetch(url, { ...options, mode: 'same-origin' });
+    };
+
+    // Early return for plain fetch (no enhancements needed)
+    if (
+      this.integrationConfig.limitPerMinute <= 0 &&
+      this.integrationConfig.maxRetries <= 0
+    ) {
+      return fetchFn;
+    }
+
+    // Apply retry wrapper if configured
+    if (this.integrationConfig.maxRetries > 0) {
+      fetchFn = this.withRetry(fetchFn);
+    }
+
+    // Apply throttling wrapper if configured
+    if (this.integrationConfig.limitPerMinute > 0) {
+      fetchFn = pThrottle({
+        limit: this.integrationConfig.limitPerMinute,
+        interval: 60_000,
+      })(fetchFn);
+    }
+
+    return fetchFn;
+  }
+
+  private withRetry(fetchFn: FetchFunction): FetchFunction {
+    return async (url, options) => {
+      let retries = 0;
+
+      while (retries <= this.integrationConfig.maxRetries) {
+        this.queue.push(() => {});
```

Comment:
This seems surprising. :)

Reply:
It might be more readable if the queue were a queue of request, and we pushed url.